spec/lib/reports/careplus_exporter_spec.rb

```diff
@@ -11,24 +11,38 @@
   end
 
   shared_examples "generates a report" do |programme_type|
+    let(:programme_type) { programme_type }
+    let(:delivery_method) { :intramuscular }
     let(:programme) { create(:programme, programme_type) }
+    let(:programme_type) { nil }
+    let(:parsed_csv) { CSV.parse(csv) }
+    let(:headers) { parsed_csv.first }
+    let(:programmes) { [programme] }
+    let(:vaccine) { create(:vaccine, programme:) }
+    let(:session) { create(:session, organisation:, programmes:, location:) }
+    let(:location) do
+      create(
+        :school,
+        gias_local_authority_code: 123,
+        gias_establishment_number: 456
+      )
+    end
+    let(:organisation) do
+      create(:organisation, careplus_venue_code: "ABC", programmes:)
+    end
+    let(:patient_session) do
+      create(
+        :patient_session,
+        :consent_given_triage_not_needed,
+        programmes:,
+        session:
+      )
+    end
+    let(:expected_vaccine_code) do
+      described_class::PROGRAMME_TYPE_TO_VACCINE_CODE.fetch(programme_type)
+    end
```

Comment:
I think this can all go inside the `context` below it (like some of these used to be) - and I think that might even allow us to go back to `includes_examples` maybe?

Reply:
Ah yeah, okay, that makes sense because the `let(:programme)` is defined outside the context.